lib/action_controller/responder.rb

```diff
@@ -129,7 +129,8 @@ class Responder
     def initialize(controller, resources, options={})
       @controller = controller
       @request = @controller.request
-      @format = @controller.formats.first
+      
+      @format = @controller.formats.detect{|f| Renderers::RENDERERS.include?(f) } || @controller.formats.first
```

Comment:
Should it be based on the renderers? I believe it should use the list of accepted mime types instead of the renderers. It is perfectly fine to accept a request that doesn't have a renderer but the controller action know how to handle it.